source/slang/slang.cpp

```diff
@@ -243,91 +243,102 @@ static SourceLanguage inferSourceLanguage(CompileRequest* request)
+        // a pass-through compilation.
+        //
+        // Note that we *do* perform output generation as normal in pass-through mode.
+        if (passThrough == PassThroughMode::None)
         {
-            parseTranslationUnit(translationUnit.Ptr());
-        }
-        if (mSink.GetErrorCount() != 0)
-            return 1;
-
-        // Perform semantic checking on the whole collection
-        checkAllTranslationUnits();
-        if (mSink.GetErrorCount() != 0)
-            return 1;
+            // Parse everything from the input files requested
+            for (auto& translationUnit : translationUnits)
+            {
+                parseTranslationUnit(translationUnit.Ptr());
+            }
+            if (mSink.GetErrorCount() != 0)
+                return 1;
 
-        // Generate initial IR for all the translation
-        // units, if we are in a mode where IR is called for.
-        generateIR();
+            // Perform semantic checking on the whole collection
+            checkAllTranslationUnits();
+            if (mSink.GetErrorCount() != 0)
+                return 1;
 
-        if (mSink.GetErrorCount() != 0)
-            return 1;
+            // Generate initial IR for all the translation
+            // units, if we are in a mode where IR is called for.
+            generateIR();
 
-        // For each code generation target generate
-        // parameter binding information.
-        // This step is done globaly, because all translation
-        // units and entry points need to agree on where
-        // parameters are allocated.
-        for (auto targetReq : targets)
-        {
-            generateParameterBindings(targetReq);
             if (mSink.GetErrorCount() != 0)
                 return 1;
+
+            // For each code generation target generate
+            // parameter binding information.
+            // This step is done globaly, because all translation
+            // units and entry points need to agree on where
+            // parameters are allocated.
+            for (auto targetReq : targets)
+            {
+                generateParameterBindings(targetReq);
+                if (mSink.GetErrorCount() != 0)
+                    return 1;
+            }
         }
-    }
-    
-    // If command line specifies to skip codegen, we exit here.
-    // Note: this is a debugging option.
-    if (shouldSkipCodegen)
-        return 0;
 
-    // Generate output code, in whatever format was requested
-    generateOutput(this);
-    if (mSink.GetErrorCount() != 0)
-        return 1;
+        // If command line specifies to skip codegen, we exit here.
+        // Note: this is a debugging option.
+        if (shouldSkipCodegen)
+            return 0;
 
-    return 0;
+        // Generate output code, in whatever format was requested
+        generateOutput(this);
+        if (mSink.GetErrorCount() != 0)
+            return 1;
+
+        return 0;
+    }
+    catch (InternalError)
+    {
```

Comment:
There is supposed to be a `catch` for these higher up, inside of `slang.cpp`. I almost always end up disabling it when debugging, though, because having actual honest-to-goodness internal error get swallowed is something we want for clients of the library, but not for us when implementing it.

At the very least I'd want an easy `#ifdef` for toggling this (which is what I did in `slang.cpp`, IIRC).

Reply:
I am fine to have a `#ifdef ` for this, but I usually use visual studio's break on exception feature for debugging purposes.

Reply:
I've worked with that before, but it definitely isn't ideal for when I'm working inside of Falcor and trying to repro issues, since I end up working with a bunch of other people's projects/executables, and it seems like the relevant settings don't "stick" between projects.

Either way we probably want only a single point where we catch this stuff, rather than two...

Reply:
I found that top level catching logic. I've fixed it with proper #ifdef.